File: CodingTest/성유진/21주차/네트워크.java
import java.util.*;
import java.io.*;

class Solution {
  boolean[] visited;
  int N;

  void bfs(int n, int[][] computers) {
    Queue<Integer> Q = new LinkedList<>();
    Q.offer(n);
    visited[n] = true;
    
    while (!Q.isEmpty()) {
      int cur = Q.poll();
      for (int i = 0; i < N; i++) {
        // 현재 노드와 인접하고 방문하지 않았는지 확인
        if (computers[cur][i] == 1 && !visited[i]) {
          visited[i] = true;
          Q.offer(i);
        }
      }
    }
  }
    
  public int solution(int n, int[][] computers) {
    N = n;
    visited = new boolean[n];
    int answer = 0;
    for (int i = 0; i < n; i++) {
       if (!visited[i]) {
         bfs(i, computers);
         answer++;
       }
    }
    return answer;
  }
}

File: CodingTest/성호/20주차/네트워크.js
function solution(n, computers) {
  let answer = 0;
  let arr = [];
  let visited = Array(n).fill(false);

  for (let i = 0; i < computers.length; i++) {
    const input = computers[i];
    let temp = [];
    for (let j = 0; j < input.length; j++) {
      if (i === j) {
        continue;
      } else {
        if (input[j] === 1) {
          temp.push(j);
        }
      }
    }
    arr.push(temp);
  }

  for (let i = 0; i < n; i++) {
    if (!visited[i]) {
      bfs(i);
    }
  }

  function bfs(i) {
    answer++;
    let q = [i];
    visited[i] = true;
    while (q.length > 0) {
      const curr = q.shift();
      for (let temp of arr[curr]) {
        if (!visited[temp]) {
          q.push(temp);
          visited[temp] = true;
        }
      }
    }
  }
  return answer;
}
